src/main/java/com/github/orest/car_shop/controllers/CarController.java

```diff
@@ -0,0 +1,114 @@
+package com.github.orest.car_shop.controllers;
+
+import com.github.orest.car_shop.model.Car;
+import com.github.orest.car_shop.repository.CarRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.FieldError;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+import org.springframework.web.servlet.view.RedirectView;
+
+import javax.validation.Valid;
+import java.util.Map;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+@RequestMapping
+@Controller
+public class CarController {
+
+    @Autowired
+    private CarRepository carRepository;
+
+    @GetMapping("/edit_car")
+    public String edit(@RequestParam int id, RedirectAttributes attributes, Model model) {
+
+        model.addAttribute("fields", Car.class.getDeclaredFields());
+        if (model.containsAttribute("car")) {
+            model.addAttribute("cars", model.getAttribute("car"));
+        } else {
+            model.addAttribute("cars", carRepository.findById(id));
+        }
+        model.addAttribute("available_text", "Car edit page:");
+
+
+        return "edit_car";
+    }
+
+    @PostMapping("update_car")
+    public String redirectIndexUpdate(@RequestParam int id, RedirectAttributes attributes, @Valid Car car, BindingResult bindingResult, Model model) {
+        if (bindingResult.hasErrors()) {
```

Comment:
I don't think that you need `bindingResult` because annotation `@Valid` will return bad request with incorrect fields. It'd be like this:
<img width="396" alt="image" src="https://user-images.githubusercontent.com/49312512/84257601-3ed90500-ab1e-11ea-8e3b-b94b6c693445.png">

Reply:
it throws page crash instead of catching error, so i'm living as it is